tests/repository_test.py

```diff
@@ -414,3 +416,12 @@ def test_local_repository():
     with pytest.raises(NotImplementedError):
         local_repo.manifest
     assert len(local_repo.hooks) == 1
+
+
+def test_norm_version_expanduser():
+    old_home = os.environ.get('HOME', None)
+    test_home = '/Users/someuser'
+    os.environ['HOME'] = test_home
+    expected_path = '/Users/someuser/.pyenv/versions/3.4.3/bin/python'
+    assert norm_version('~/.pyenv/versions/3.4.3/bin/python') == expected_path
+    os.environ['HOME'] = old_home
```

Comment:
you'll want to use:

```python

with mock.patch.dict(os.environ, {'HOME': test_home}):
    # run your assertion
```

right now when the assertion fails, the environment is polluted for the rest of the test.

I think you might also need to twiddle the file paths so the test passes on windows.